src/charm.py

```diff
@@ -622,6 +629,104 @@ def _get_cpu_extensions() -> list[str]:
                 del cpu_flags[0]
         return cpu_flags
 
+    def _get_access_nad_config(self) -> Dict[Any, Any]:
+        """Get access interface NAD config.
+
+        Returns:
+            config (dict): Access interface NAD config
+
+        """
+        config = {
+            "cniVersion": "0.3.1",
+            "ipam": {
+                "type": "static",
+                "routes": [
+                    {
+                        "dst": self._get_gnb_subnet_config(),
+                        "gw": self._get_access_network_gateway_ip_config(),
+                    },
+                ],
+                "addresses": [
+                    {
+                        "address": self._get_access_network_ip_config(),
+                    }
+                ],
+            },
+            "capabilities": {"mac": True},
+        }
+        if self._get_access_interface_mtu_config():
+            config.update({"mtu": self._get_access_interface_mtu_config()})  # type: ignore
+        return config
+
+    def _get_core_nad_config(self) -> Dict[Any, Any]:
+        """Get core interface NAD config.
+
+        Returns:
+            config (dict): Core interface NAD config
+
+        """
+        config = {
+            "cniVersion": "0.3.1",
+            "ipam": {
+                "type": "static",
+                "addresses": [
+                    {
+                        "address": self._get_core_network_ip_config(),
+                    }
+                ],
+            },
+            "capabilities": {"mac": True},
+        }
+        if self._get_core_interface_mtu_config():
+            config.update({"mtu": self._get_core_interface_mtu_config()})  # type: ignore
+        return config
+
+    def _get_core_interface_mtu_config(self) -> Optional[int]:
+        """Get Core interface MTU size.
+
+        Returns:
+            mtu_size (int/None): If MTU size is not configured return None
+                                    If it is set, returns the configured value expected in int
+
+        """
+        return self.model.config.get("core-interface-mtu-size")  # type: ignore
+
+    def _get_access_interface_mtu_config(self) -> Optional[int]:
+        """Get Access interface MTU size.
+
+        Returns:
+            mtu_size (int/None): If MTU size is not configured return None
+                                    If it is set, returns the configured value expected in int
+
+        """
+        return self.model.config.get("access-interface-mtu-size")  # type: ignore
+
+    def _access_interface_mtu_size_is_valid(self) -> bool:
+        """Checks whether the access interface MTU size is valid.
+
+        Returns:
+            bool: Whether access interface MTU size is valid
+        """
+        if self._get_access_interface_mtu_config() is None:
+            return True
+        try:
+            return self._get_access_interface_mtu_config() in range(1200, 65537)
+        except ValueError:
+            return False
```

Comment:
This is more efficient:

```suggestion
        return (1200 <= self._get_access_interface_mtu_config() < 65537)
```

Reply:
def _get_core_interface_mtu_config(self) -> Optional[int]:
        return self.model.config.get("core-interface-mtu-size")  

I tested the return value before writing Optional[int]. If we set int, here the return type was int. Anyway in the config.yaml I set the value type as int and it does not allow me to set a string.

Reply:
You are suggesting to set Optional[str] to fix type for below code block.  If we will have type problem in the following line which is type str could not be compared in that way.
 (1200 <= self._get_access_interface_mtu_config() < 65537)
 ```
 def _get_access_interface_mtu_config(self) -> Optional[int]:
        return self.model.config.get("access-interface-mtu-size") 
 ```

Reply:
I agree that we will need to do an explicit `int` conversion, but it happened anyway implicitly before.

Reply:
Returned as str, then str -> int validation is done in validation.